### PR TITLE
[fga] more FGA checks and service use

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 3ac3eb7</samp>

This pull request implements the user deletion feature and refactors some of the server components to improve authorization and code quality. It adds a new `delete` permission to the `UserPermission` type and the `User` schema in `SpiceDB`, and uses the `Authorizer` and the `WorkspaceService` for permission checks and workspace operations. It also adds new tests for the `WorkspaceService` methods and refactors some of the existing code to use dependency injection and remove redundancy.

</details>

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -1057,9 +1058,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const ws = await this.workspaceService.getWorkspace(user.id, workspaceId);
         await this.guardAccess({ kind: "workspace", subject: ws }, "delete");
 
-        // for good measure, try and stop running instances
-        await this.internalStopWorkspace(ctx, user.id, ws, "deleted via API");
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:1061]: @geropl I deleted this here because it happens as part of deleteWorkspace below.

Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:1061]: Only difference I see is the "delete via API" is replaced by "deleted via WorkspaceService". But not sure how we used that anyway. :+1:

**Comment by geropl** [comment]
Taking a closer look now :eyes:

**Comment by geropl** [comment]
@svenefftinge Would be awesome if you could merge this soon so I can rebase :rocket: :nerd_face: :pray:

**Pull request merged**